desci-server/src/controllers/doi/check.ts

```diff
@@ -31,17 +37,81 @@ export const checkMintability = async (req: RequestWithNode, res: Response, _nex
   }
 };
 
-export const getDoi = async (req: Request, res: Response, next: NextFunction) => {
-  const { identifier } = req.params;
+interface WorksDetails {
+  doi: string;
+  authors: string[];
+  citation_count: number;
+  pdf_url: string;
+  publication_year: string;
+  works_id: string;
+  work_type: string;
+  title: string;
+  landing_page_url: string;
+}
+
+export const retrieveDoi = async (req: Request, res: Response, _next: NextFunction) => {
+  const { doi: doiQuery, uuid, dpid } = req.query;
+  const identifier = doiQuery || uuid || dpid;
+
   if (!identifier) throw new BadRequestError();
 
-  const pending = await doiService.hasPendingSubmission(ensureUuidEndsWithDot(identifier));
-  logger.info({ pending }, 'GET DOI');
-  if (pending) {
-    new SuccessResponse({ status: pending.status }).send(res);
-    return;
+  if (uuid) {
+    const pending = await doiService.hasPendingSubmission(ensureUuidEndsWithDot(uuid as string));
+    logger.info({ pending }, 'GET DOI');
+    if (pending) {
+      new SuccessResponse({ status: pending.status }).send(res);
+      return;
+    }
   }
 
-  const doi = await doiService.getDoiByDpidOrUuid(identifier);
-  new SuccessResponse(doi).send(res);
+  const doiLink = (doiQuery as string).startsWith('https') ? doiQuery : `https://doi.org/${doiQuery}`;
+
+  const client = new Client({
+    connectionString: process.env.OPEN_ALEX_DATABASE_URL,
```

Comment:
add a .env.example line for this and update the k8s configs @shadrach-tayo

Reply:
oh damn missed that